lib/index.js

```diff
@@ -166,130 +171,115 @@ function GracefulShutdown(server, opts) {
     }
   });
 
-  process.on('exit', function () {
+  process.on('close', function () {
     debug('closed');
   });
 
   // shutdown event (per signal)
   // ----------------------------------
   function shutdown(sig) {
+    debug('shutdown signal - ' + sig);
 
-    return new Promise((resolve, reject) => {
+    function cleanupHttp() {
+      destroyAllConnections();
+      debug('Close http server');
 
-      function cleanupHttp() {
-        return new Promise((resolve, reject) => {
-          destroyAllConnections();
+      return new Promise((resolve, reject) => {
+        server.close((err) => {
+          if (err) {
+            return reject(err);
+          }
 
-          // normal shutdown / do not allow new connections
-          server.close((err) => {
-            if (err) { return reject(err); }
-            resolve();
-          });
+          return resolve();
         });
+      });
+    }
+
+    // Don't bother with graceful shutdown on development to speed up round trip
+    if (options.development) {
+      debug('DEV-Mode - immediate forceful shutdown');
+      return process.exit(0);
+    }
```

Comment:
thinking out loud: now that there is a `preShutdown` handler, this behavior could be moved to that — something I bet the library author would like.

Reply:
Love this idea. Moved the dev mode shut down into preShutdown.

Reply:
hm... I ended up reverting it. This would be really nice to have if your next suggestion of `newConnectionsAreStillBeingMade`  is straightforward to implement. Ideally, the user of the lib does not need to worry about overwriting the dev mode behavior.